post_processing/orac_postproc.F90

```diff
@@ -329,6 +341,15 @@ subroutine orac_postproc(mytask, ntasks, lower_bound, upper_bound, &
        case('CORRECT_PARALLAX')
           if (parse_string(value, corr_plx) /= 0) &
                call handle_parse_error(label)
+       case('BAYESIAN_WEIGHTS')
+          ! Note that we expect the bayesian_weights array to have more elements
+          ! than given in the driver file!
+          stat = parse_string(value, bayesian_weights)
```

Comment:
Do we want to check that this is the same length as the number of files provided? It seems easy is accidentally add/subtract a single number from a 20 element list and unintentionally delete an aerosol type.

Reply:
Is there any easy way of getting parse_string to tell you how many values it read? If not, this is not actually entirely trivial to do. If you did get the list of weights wrong, you wouldn't delete an aerosol type, since the weights default to 1, but you would have the wrong weights (obviously!)

Reply:
Oh, wait, actually that's a bug (I didn't test this keyword).

`parse_string` takes an optional argument `n_in`, which tells it how long the array it reads should be. If that argument isn't given, the desired length is assumed to be length of the input array. In this case, that's `MaxInFiles` which won't be appropriate.

I think this needs to be,
```
stat = parse_string(value, bayesian_weights, n_in=n_in_files)
```
or
```
stat = parse_string(value, bayesian_weights(1:n_in_files))
```
whichever you think is easier to understand.